### ISSUE TITLE
Running example unit tests behind corporate proxy

### ISSUE THREAD

**Issue by Jorrit05** [MicrosoftDocs/vsts-docs#7250]
I am running the example unit tests behind a corporate proxy. It tries to download a tar file:
  Sample task tests
Downloading file: https://nodejs.org/dist/v10.15.1/node-v10.15.1-darwin-x64.tar.gz

But it is not picking up the environment proxy settings (https_proxy and http_proxy), so is there anyway to do this and could it be added to the example?
Thanks

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 44581f4a-93df-7507-3844-eadedf9a6cc4
* Version Independent ID: 33c7bcf0-1512-975f-5ef8-9f2a192a84ad
* Content: [Add a Build or Release Task - Azure DevOps](https://docs.microsoft.com/en-us/azure/devops/extend/develop/add-build-task?view=azure-devops#feedback)
* Content Source: [docs/extend/develop/add-build-task.md](https://github.com/MicrosoftDocs/vsts-docs/blob/master/docs/extend/develop/add-build-task.md)
* Product: **devops**
* Technology: **devops-ecosystem**
* GitHub Login: @chcomley
* Microsoft Alias: **chcomley**

**Comment by WilliamAntonRohm** [comment]
@Jorrit05 -- thank you for your feedback.  You may find answers here:

- [Azure DevOps Support Bot](https://azuredevopsvirtualagent.azurewebsites.net/) 
- [Azure DevOps on Stack Overflow](https://stackoverflow.com/questions/tagged/azure-devops) 

@chcomley -- please look into this issue.

**Comment by chcomley** [comment]
Hi @Jorrit05, Please see [Run a self-hosted agent behind a web proxy](https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/proxy?view=azure-devops&tabs=windows). Thank you!

**Comment by r-or** [comment]
Hi @ chcomley, I believe this doesn't fix the issue. mock-test.ts tries to download a specific nodejs version. If I want to test my task locally, as it is done in this tutorial, the download will fail.

I tried to fix the issue (make it pick up on http_proxy) and I had to replace sync-request with a custom made request call to be able to insert a proxy agent.

It's a bit cobbled together, but if anyone's interested, here is the fork: 
https://github.com/r-or/azure-pipelines-task-lib/tree/proxy2

It picks up the "http_proxy" environment variable.
